tests/integration/setup.coffee

```diff
@@ -84,8 +84,8 @@ exports.resetUser = ->
 
 exports.credentials = buildCredentials()
 
-exports.givenLoggedInUserWithApiKey = ->
-	beforeEach ->
+exports.givenLoggedInUserWithApiKey = (beforeFn = beforeEach) ->
```

Comment:
This might cause some confusion down the road, so maybe it is better to be explicit about the function we are passing (instead of a default value). I know it is a bit more verbose, but it is immediately obvious from the tests whether it runs on before or beforeEach. I don't have a strong opinion on it, so I will leave it up to you to decide.

Reply:
Totally agree on that. I initially used a default value, since I was originally thinking about just changing the device.spec, which was only of the larger parts of our test suit.
Since this escalated and touched most of the test cases, it totally makes sense to make this mandatory and throw if missing. Will do :+1: